datamanager/pumper/pumper.go

```diff
@@ -0,0 +1,103 @@
+package pumper
+
+import (
+	"context"
+	"errors"
+	"github.com/bitleak/lmstfy/config"
+	"github.com/bitleak/lmstfy/datamanager/storage"
+	"github.com/bitleak/lmstfy/helper"
+	"github.com/go-redis/redis/v8"
+	"github.com/go-redsync/redsync/v4/redis/goredis/v8"
+	"github.com/sirupsen/logrus"
+	"time"
+
+	"github.com/bitleak/lmstfy/datamanager/storage/model"
+	"github.com/bitleak/lmstfy/engine"
+	"github.com/go-redsync/redsync/v4"
+)
+
+const (
+	DefaultStoragePumpPeriod = 10 * 60 // number of seconds, equals 10 minutes
+	LoopPumpLockName         = "LP_redsync"
+	MaxJobBatchSize          = 1000
+)
+
+type DefaultPumper struct {
+	redisClient       *redis.Client
+	shutdown          chan struct{}
+	storagePumpPeriod int64
+}
+
+func NewPumper(cfg *config.Config, poolConfig *config.RedisConf) (Pumper, error) {
```

Comment:
NewPumper => NewDefaultPumper

Reply:
For `NewDefaultPumper`, I think the below sign will be better :

`NewPumper(cfg *config.Config, stor storage.Storage, eng engine.Engine)` then remove the `storage` and `engine` from `LoopPump`, which make parameters more orthogonal. I felt a bit strange at my first glance since the global config contained the poolConfig.

Reply:
Yes, will fix this part.